### PR TITLE
Sort architectural characteristics based on the order of importance + update sequence diagram

### PR DESCRIPTION
**Pull request by samira-barouti**
* Sort architectural characteristics based on the order of importance
* Add legends to the sequence diagram
* Add a new sequence for new service notification when the non-profit is registered

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,19 +201,23 @@ Generally, these would include all the infrastructure, cloud, services, licenses
...
+The purpose of this product is to facilitate collaboration between non-profits. To achieve this goal, Spotlight needs to communicate with external systems (such as childcare/housing services) and legacy systems.
 
-### h. Availability
+### g. Data Integrity
 
-Amazon provides services in multiple Availability Zones. By using Serverless services, the Diversity Cyber Council platform will make use of these, by default and as such, the platform can withstand the loss of a given Availability Zone.
+The candidate's career path and progress, as well as non-profit information, are accessed by multiple personas. It is crucial to ensure the data is accessed by the authorized person, and the accuracy of the data is maintained throughout its lifecycle.
 
-### j. Scalability
+### i. Scalability
```
Comment (scott-lew) [inline README.md:220]: So I was thinking... shouldn't our solution be elastic? I don't ever see an increasing amount of load - i see a more burst type load that would happen after work hours. Maybe we can elaborate on this and compare to load patterns at other course sites or job search sites?

Comment (samira-barouti) [reply README.md:220]: um I did not remove scalability. My main focus for this MR was on the top 3: Usability, Responsiveness, Feasibility.
Regarding the scalability
I am seeing main flows: non-profit/candidate registration (write to DB) AND non-profit search (mostly read operations).
The first flow is done once when the non-profit/candidate registers, and some steps are asynchronous. I don't expect too much traffic here. Viewing the content (pdfs, rich text) is not going to put too much load on the system if we are using CDN (noticed that we don't consider CDN in our architecture). However querying non-profits might impose load on the system. 
All in all, I agree scalability is important but it is not probably in the top 3. That's my 2 cents :)
PS: we need to spend a few minutes as a team sorting the remaining 5 characteristics. I identified the top 3 (it is open to discussion for sure :D), but the rest are not sorted.

Comment (scott-lew) [reply README.md:220]: Oh, no I know you didn't remove scalability. My comment was basically maybe we should remove it and replace it with elasticity instead. I agree let's chat in a meeting re: these characteristics

Comment (MrMarc) [reply README.md:220]: Note to us, remove the "if selected ..." text

Comment (MrMarc) [reply README.md:220]: For each characteristic that we do choose, we should block-quote the text in the requirements that the characteristic was infered from.

**Review by donalddonovan — APPROVED** [review]
I dont see any issues with the PR in its current state.

**Comment by samira-barouti** [comment]
I'll merge this. We can revisit the order later, if needed.

**Pull request merged**